src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,76 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String PARSED_STRING_SEPARATOR = ";";
+    private static final String FORMED_STRING_SEPARATOR = ",";
+    private static final String BUY_CATEGORY_NAME = "buy";
+    private static final String SUPPLY_CATEGORY_NAME = "supply";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] data = readDataFromFile(fromFileName);
+        String report = formReport(data);
+        createNewDataFile(toFileName, report);
+    }
+
+    public String[] readDataFromFile(String fromFileName) {
+        try (FileReader fileReader = new FileReader(fromFileName)) {
+            StringBuilder parsedCsv = new StringBuilder();
+            BufferedReader reader = new BufferedReader(fileReader);
```

Comment:
```suggestion
         StringBuilder parsedCsv = new StringBuilder();
        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
```
1) you should keep the least amount of logic in the try block that can cause an exception 
2) BufferedReader extends from Reader that implements Closeable. It’s mean that you should  use it in a try-with-resources block